### ISSUE TITLE
Capsule Network example training

### ISSUE THREAD

**Issue by DanielDimanov** [ageron/handson-ml#543]
First I want to take the chance to thank you for your videos, tutorials, books and for publishing the notebooks in a public repo! Thank you! 

I recently started looking at Capsule Networks and found your tutorials extremely useful! However, when I tried to run the notebook everything runs normally, no errors or anything, the loss seems to be low, but the accuracy is just around 9% and when I try to run the reconstruction example I get just random blobs, so I was wondering if it is a common problem or is it just me? I tried deleting everything and cloning the repo again and getting a new Anaconda environment and got the same results. Do you know what the reason for this may be and how can I resolve it? Probably there is something up with the loss calculation, but to be fair thre shouldn't be.. I'm attaching screenshots of the reconstructions and the caps output vectors. 
![reconstruct-1](https://user-images.githubusercontent.com/17221902/72441160-18e34a80-37a2-11ea-857f-690c1994f48b.png)
![train](https://user-images.githubusercontent.com/17221902/72441162-18e34a80-37a2-11ea-9b25-2d21897240bc.png)
![visualisation](https://user-images.githubusercontent.com/17221902/72441163-197be100-37a2-11ea-979c-5ff4b1b484f0.png)

Any help will be appreciated!

**Comment by warnickr** [comment]
I see this same issue when I run the notebook on GPUs. It runs fine on my MacBook CPU

**Comment by bfromson** [comment]
I'm seeing this issue on my desktop box running Ubuntu 18.04 with python and tensorflow installed using anaconda.

Running with GPU on Colab it runs fine.

Looking at data flowing through the first 5 items of the first batch give identical results on both environments and after that the differ. If I run a batch_size of 5 on my desktop then it runs fine, batch_size 6 and it is wrong. Trying to find out why, but not there yet.

**Comment by bfromson** [comment]
Problem is due to calling tensorflow.matmul with more than 2**16 matrix multiplies. It is a bug in CUDA 10.0 but fixed in 10.1. See https://github.com/tensorflow/tensorflow/issues/32877
